templates/hammer_root.yml.epp

```diff
@@ -0,0 +1,8 @@
+<%- |
+  Optional[String] $username,
+  Optional[String] $password,
+| -%>
+:foreman:
+  # Credentials. You'll be asked for the interactively if you leave them blank here
+  :username: '<%= if $username { $username } else { 'undef'} %>'
+  :password: '<%= if $password { $password } else { 'undef'} %>'
```

Comment:
I'm not sure if `undef` is really intended. You're probably right that that was the previous behavior but it wasn't really intended. IIRC Puppet < 4 evaluated it to an empty string and we didn't have a test for it. I doubt hammer handles `undef`.

tl;dr: I think it should be this:

```suggestion
  :username: '<%=$username %>'
  :password: '<%= $password %>'
```

Reply:
Suggestion implemented.